src/Q10/kyunghee/Solution.java

```diff
@@ -0,0 +1,46 @@
+package problems;
+
+public class Solution {
+    private int[][] directions = {
+            {1, 0},  //상
+            {-1, 0}, //하
+            {0, -1}, //좌
+            {0, 1}   //우
+    };
+
+    public int numIslands(char[][] grid) {
+        int count = 0;
+        for (int y = 0; y < grid.length; y++) {
+            for (int x = 0; x < grid[0].length; x++) {
+                if (grid[y][x] == '1'){
+                    count++;
+                    dfs(grid, y, x);
+                }
+            }
+        }
+
+        return count;
+    }
+
+    private void dfs(char[][] grid, int y, int x) {
+        if (grid[y][x] == '2') {    //visited
+            return;
+        }
+
+        for (int i = 0; i < directions.length; i++) {
+            if (check(grid, y+directions[i][0], x+directions[i][1])) {
+                grid[y][x] = '2';
+                dfs(grid, y+directions[i][0], x+directions[i][1]);
+            }
+        }
+    }
+
+    private boolean check(char[][] grid, int y, int x) {
+        if (y < 0 || grid.length == y || x < 0 || grid[0].length == x) { // edge
+            return false;
+        } else if (grid[y][x] == '0') { // water
+            return false;
+        }
+        return true;
```

Comment:
```suggestion
        }        
        return grid[y][x] == '1';
```'
이렇게 하면 더 줄일 수도 있을 것 같네요!

Reply:
이렇게 하면 조금 더 줄일 수도 있을 것 같습니다!

Reply:
check 메소드 명도 약간 더 구체적으로 적으면 더 명확할것같아요
예를들면 isIsland?

Reply:
좋은 의견 감사합니다 isIsland 혹은 checkIsland 정도면 괜찮을 것 같습니다

Reply:
> ```suggestion
> 
>         }        
> 
>         return grid[y][x] == '1';
> 
> ```

@JungHyunLyoo 정현님 코드를 보니 isIsland에 반환 결과와 같군요 간결하고 명시적입니다!
감사합니다